pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java

```diff
@@ -139,4 +141,22 @@ protected void cleanup() throws Exception {
                 .enableBatchIndexAcknowledgment(true)
                 .subscribe();
     }
+
+    @Test
+    public void testGetTxnID() throws Exception {
+        Awaitility.await().atMost(4, TimeUnit.SECONDS).until(()->{
+            try {
+                Transaction transaction = pulsarClient.newTransaction()
```

Comment:
@congbobo184 @codelipenghui 
I believe that this is not a good expected behaviour for a client.

it should be the newTransaction() method that waits for one Coordinator to be available up to a given (configurable) timeout.
otherwise the API is difficult to understand and to use.



can we add an utility method in TransactionTestBase like "waitForCoordinatorToBeAvailable" ?

Reply:
Very good suggestion.
but this PR is mainly for getTxnID, and waitForCoordinatorToBeAvailable will be optimized later

Reply:
I understand your concern @liangyuanpeng 
But if we leave here this Awaiatilty loop we won't remove it in the future probably.

It is better to fix it now

Reply:
If you prefer we can create a separate patch to be committed before this patch.
But adding that method should be easy

Reply:
Different tests may require different numbers of boker and tc.
 Is it more appropriate to mention a pr to fix this problem ?
 If you don't mind, I will do this later.

Reply:
I have optimized waitForCoordinatorToBeAvailable method for TransactionTest.